app/main/helpers/suppliers.py

```diff
@@ -27,3 +27,24 @@ def get_country_name_from_country_code(country_code):
 
 
 COUNTRY_TUPLE = load_countries()
+
+
+def supplier_company_details_are_complete(supplier_data):
+    supplier_required_fields = ['dunsNumber', 'name', 'registeredName', 'registrationCountry', 'registrationDate',
+                                'organisationSize', 'tradingStatus']
+    contact_required_fields = ['address1', 'city', 'postcode']
+
+    registration_country = supplier_data['registrationCountry'] if 'registrationCountry' in supplier_data else None
+    if registration_country == 'gb' or registration_country == 'country:GB':
+        supplier_required_fields.append('companiesHouseNumber')
+        supplier_required_fields.append('vatNumber')
+
+    else:
+        supplier_required_fields.append('otherCompanyRegistrationNumber')
+
+    return (
+        all([f in supplier_data and supplier_data[f] for f in supplier_required_fields]) and
```

Comment:
Hi, what does this `f in supplier_data and supplier_data[f]` syntax mean in this list comprehension?

Reply:
`f in supplier_data` means that the key `f` must be present in `supplier_data`. `supplier_data[f]` means that the value pointed at by `f` in `supplier_data` must be Truthy.

Reply:
Ahh, ok, makes sense, thank you! I guess it's just first time I've seen list comprehensions used for returning booleans :)

Reply:
Do you think it might be useful to change "f" to "key"?

Reply:
I feel like `f` is alright, as a shorthand for `field` that's referenced in the main variable of the list comprehension. Generally in comprehensions I think it's ok (and personally I prefer) to use one-letter variables.